middleware/service/source/manager/transform.cpp

```diff
@@ -120,6 +120,26 @@ namespace casual
                };
             }
 
+            auto reservation()
+            {
+               return []( auto& pair)
+               {
+                  auto& instance = std::get< 1>( pair);
+
+                  manager::admin::model::Reservation result;
+                  result.service = instance.reserved_service().value_or( result.service);
```

Comment:
Is this a good paradigm? I had to look an extra time to understand that it is just _value or default_. 
Hmm, it gets ugly to do it any other way...
We should have an _algorithm_ for this. like `coalesce` but returns `T` default initialised if all _optional like_ is empty.